### PR TITLE
feat(console): Added command getaddrinfo, set/get dnsserver to console_cmd_ping

### PR DESCRIPTION
**Pull request by espressif-abhikroy**
Added command getaddrinfo, setdnsserver, getdnsserver to console_cmd_ping component.
Removed WHOLE_ARCHIVE from CMakelist, and allows users to disable auto-registration of console_cmd_ping component.

### CODE REVIEW

File: components/console_cmd_ping/Kconfig.projbuild [hunk 1]
```diff
@@ -0,0 +1,21 @@
+menu "Ping command Configuration"
+
+    config PING_CMD_AUTO_REGISTRATION
```
Comment (david-cermak) [inline components/console_cmd_ping/Kconfig.projbuild:3]: to simplify this, maybe we can support only one auto-registration config per component?

File: components/console_cmd_ping/README.md [hunk 2]
```diff
@@ -29,13 +29,18 @@ The component provides a console where the 'ping' command can be executed.
 
     // To register only ifconfig command skip calling console_cmd_all_register()
```
Comment (david-cermak) [inline components/console_cmd_ping/README.md:30]:
```suggestion
    // To register only ping/dns commands skip calling console_cmd_all_register()
```

File: components/console_cmd_ping/console_getsetdnsserver.c [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+ * This function iterates over all network interfaces available on the ESP32 device
+ * and sets the DNS server address for the specified DNS type (main, backup, or fallback).
+ * The DNS address is only set if a valid address is provided (non-zero and not equal to IPADDR_NONE).
+ *
+ * @param addr IP address of the DNS server.
+ * @param type Type of the DNS server (main, backup, fallback).
+ *
+ * @return esp_err_t Returns ESP_OK on success, or an error code on failure.
+ */
+static esp_err_t set_dns_server(uint32_t addr, esp_netif_dns_type_t type)
```
Comment (david-cermak) [inline components/console_cmd_ping/console_getsetdnsserver.c:74]: do we support only IPv4 addresses?

**Review by david-cermak — APPROVED** [review]
LGTM, just left few nitpicks

File: components/console_cmd_ping/Kconfig.projbuild (hunk 1 above)
Comment (espressif-abhikroy) [reply components/console_cmd_ping/Kconfig.projbuild:3]: Done

File: components/console_cmd_ping/console_getsetdnsserver.c (hunk 3 above)
Comment (espressif-abhikroy) [reply components/console_cmd_ping/console_getsetdnsserver.c]: The initial implementation was only for ipv4. Updated to use ipv6 also.
Now this is supported
`setdnsserver 8.8.8.8 fe80::b0be:83ff:fe77:dd64 www.xyz.com`

**Pull request merged**